receipt_listener.go

```diff
@@ -20,23 +20,30 @@ type ReceiptListener struct {
 	log      zerolog.Logger
 	provider *ethrpc.Provider
 	monitor  *ethmonitor.Monitor
+	br       *breaker.Breaker
 
 	pastReceipts []BlockOfReceipts
-	subscribers  []*subscriber
 
-	mu sync.Mutex
+	subscribers   []*subscriber
+	muSubscribers sync.Mutex
 }
 
 type ReceiptResult struct {
 	MetaTxnID  MetaTxnID
-	Status     MetaTxnStatus
+	Results    []*MetaTxnResult
 	TxnReceipt *types.Receipt
 }
 
+type MetaTxnResult struct {
+	Status MetaTxnStatus
+	Reason string
+}
+
 type BlockOfReceipts []*ReceiptResult
 
 type subscriber struct {
-	ch          chan *ReceiptResult
```

Comment:
I think for ch and sendCh we can use type `ReceiptResult` instead of `*ReceiptResult` .. i think contiguous memory will be more efficient, and all data types in this struct are pointers anyways, so avoid pointer to pointers -- what do you think?